### PR TITLE
PLT-3566 Summized resources for running Marlowe Runtime.

### PR DESCRIPTION
**Pull request by bwbush**
This document summarizes the information benchmarking of the three public networks. See https://github.com/input-output-hk/marlowe-cardano/blob/PLT-3566/marlowe-runtime/doc/resources.md.

Pre-submit checklist:
- Branch
    - [ ] Tests are provided (if possible)
    - [ ] [Test report is updated](https://github.com/input-output-hk/marlowe-cardano/blob/main/marlowe/test/test-report.md) (if relevant)
    - [x] Commit sequence broadly makes sense
    - [x] Key commits have useful messages
    - [x] Relevant tickets are mentioned in commit messages
    - [x] Formatting, PNG optimization, etc. are updated
- PR
    - [x] Self-reviewed the diff
    - [x] Useful pull request description
        - Review required
        - [x] Substantial changes to code, test, or documentation
        - [ ] Change made to Marlowe validator (@bwbush and @palas must be included as reviewers)
        - Review not required
        - [ ] Minor changes to non-critical code, documentation, nix derivations, configuration files, or scripts
        - [ ] Formatting, spelling, grammar, or reorganization
    - [x] Reviewer requested

### CODE REVIEW

File: marlowe-runtime/doc/resources.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+1. Indexing the blockchain from genesis to the tip.
+2. Running 500 Marlowe contracts in parallel at the maximum possible transaction-building and -submission rate.
+3. Querying the history of all Marlowe contracts at the maximum rate.
+
+| Service                 | Memory  | CPU (cores) |
+|-------------------------|--------:|------------:|
+| `marlowe-chain-indexer` | 2500 MB |        50 % |
+| `marlowe-chain-sync`    |  100 MB |        50 % |
+| `marlowe-indexer`       |  100 MB |        10 % |
+| `marlowe-sync`          |  500 MB |        50 % |
```
Comment (bwbush) [inline marlowe-runtime/doc/resources.md:19]: Actually, I just ran eight copies of `marlowe-finder` and that pushes `marlowe-sync`'s memory usage above 4 GB. I'm investigating further in order to understand its scaling behavior.

File: marlowe-runtime/doc/resources.md [hunk 2]
```diff
@@ -0,0 +1,55 @@
+# Recommended Resources for Deploying Marlowe Runtime
+
+This document summarizes recommendations for the computational resources required to deploy Marlowe Runtime on public networks (`mainnet`, `preprod`, and `preview`).
+
```
Comment (joseph-fajen) [inline marlowe-runtime/doc/resources.md:4]: I suggest we add this as a cross-reference: 

For more information about how to get started using the Cardano testnets, please see [Getting started](https://docs.cardano.org/cardano-testnet/getting-started).

File: marlowe-runtime/doc/resources.md [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+The table below lists the maximum size of the Marlowe Runtime database schemas for the three public networks, as of March 2023. Because the public blockchains grow in size over time, disk resources will increase correspondingly.
+
+| Schema    | `mainnet` | `preprod` | `preview` |
+|-----------|----------:|----------:|----------:|
+| `chain`   |    200 GB |     10 GB |      2 GB |
+| `marlowe` |      2 MB |      5 MB |    300 MB |
+| `sqitch`  |    200 kB |    200 kB |    200 kB |
+| Total     |    200 GB |     10 GB |      3 GB |
+
+PostgreSQL memory and CPU requirements depend upon the particular PostgreSQL installation. During indexing, the SQL queries for Marlowe Runtime typically require 100% of a CPU core. A small PostgreSQL installation could use a little as 2 GB of memory, but such an installation would be slow to index the blockchain, though it would be adequate for queries once the blockchain was fully indexed.
```
Comment (joseph-fajen) [inline marlowe-runtime/doc/resources.md:37]: Itty bitty typo: 

A small PostgreSQL installation could use a[s] little as ...

File: marlowe-runtime/doc/resources.md [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+
+For fast indexing, about 8 GB of memory is ample for a PostgreSQL instance with the following custom configuration.
+
```
Comment (joseph-fajen) [inline marlowe-runtime/doc/resources.md:40]: Recommend "configuration:"

**Review by joseph-fajen — APPROVED** [review]
Just a couple of comments.

File: marlowe-runtime/doc/resources.md (hunk 1 above)
Comment (bwbush) [reply marlowe-runtime/doc/resources.md:19]: Here is the situation with `marlowe-sync`.
marlowe-sync

Comment (bwbush) [reply marlowe-runtime/doc/resources.md:19]: We verified that this is an artifact of too-verbose logging and is not an intrinsic limitation of `marlowe-sync`.

File: marlowe-runtime/doc/resources.md (hunk 4 above)
Comment (bwbush) [reply marlowe-runtime/doc/resources.md:42]: Fixed in 12ffeb4d965e98f5ea3afc0a1ec54c7d3ce5e840.

File: marlowe-runtime/doc/resources.md (hunk 3 above)
Comment (bwbush) [reply marlowe-runtime/doc/resources.md]: Fixed in 12ffeb4d965e98f5ea3afc0a1ec54c7d3ce5e840.

File: marlowe-runtime/doc/resources.md (hunk 2 above)
Comment (bwbush) [reply marlowe-runtime/doc/resources.md:4]: Fixed in 12ffeb4d965e98f5ea3afc0a1ec54c7d3ce5e840.

**Pull request merged**